sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -622,6 +654,14 @@ object ScalaReflection extends ScalaReflection {
           dataType = ObjectType(udt.getClass))
         Invoke(obj, "serialize", udt, inputObject :: Nil)
 
+      case t if isValueClass(t) =>
+        val (name, underlyingType) = getConstructorParameters(t).head
```

Comment:
Can we use `getUnderlyingTypeOf` consistently? Let `getUnderlyingTypeOf` return both parameter name and type.

Or just use `getConstructorParameters` and get rid of `getUnderlyingTypeOf`?

Reply:
you're right, I should return both name and type from `getUnderlyingTypeOf`